src/cplscheme/BaseCouplingScheme.cpp

```diff
@@ -940,7 +923,11 @@ double BaseCouplingScheme::getWindowStartTime() const
 
 double BaseCouplingScheme::getWindowEndTime() const
 {
-  return getWindowStartTime() + getTimeWindowSize();
+  if (hasTimeWindowSize()) {
+    return _time.time() + _time.untilWindowEnd(_timeWindowSize);
+  } else {
+    return _time.time() + _time.untilEnd();
+  }
```

Comment:
The old implementation was problematic for, for example, `precice.Integration/Serial/Time/Explicit/SerialCoupling/DoNonfittingWindows`. I think we should factor this change out into a different PR.

Reply:
Fixed in #2213